src/Solver/Dynamic/CMakeLists.txt

```diff
@@ -5,12 +5,23 @@
 #    - Cameron Rutherford <cameron.rutherford@pnnl.gov>
 #]]
 
-gridkit_add_library(solvers_dyn
-  SOURCES
-    Ida.cpp
-  LINK_LIBRARIES
-    PUBLIC SUNDIALS::nvecserial
-    PUBLIC SUNDIALS::idas
-    PUBLIC SUNDIALS::sunlinsolklu
-  OUTPUT_NAME
-    gridkit_solvers_dyn)
+if (GRIDKIT_ENABLE_SUNDIALS_SPARSE)
+  gridkit_add_library(solvers_dyn
+    SOURCES
+      Ida.cpp
+    LINK_LIBRARIES
+      PUBLIC SUNDIALS::nvecserial
+      PUBLIC SUNDIALS::idas
+      PUBLIC SUNDIALS::sunlinsolklu
+    OUTPUT_NAME
+      gridkit_solvers_dyn)
+else()
+  gridkit_add_library(solvers_dyn
+    SOURCES
+      Ida.cpp
+    LINK_LIBRARIES
+      PUBLIC SUNDIALS::nvecserial
+      PUBLIC SUNDIALS::idas
+    OUTPUT_NAME
+      gridkit_solvers_dyn)
+endif()
```

Comment:
This makes me wonder if `gridkit_add_library` macro really helps simplify the cmake code. In this case you have to replicate the entire call to the macro. One cannot do
```cmake
target_link_library(solvers_dyn PUBLIC SUNDIALS::sunlinsolklu)
```
after the call to the macro without linking to `sunlinsolklu`.